### PR TITLE
PHP7: Proxy generator to handle return type

### PR DESCRIPTION
**Pull request by wivern-co-uk**
Hi, it's initial PR to raise a discussion and make ProxyGenerator works with php7 return types

### CODE REVIEW

**Comment by Ocramius** [comment]
@wivern-co-uk much appreciated, but we'd still require test cases to validate this logic :-)

**Comment by wivern-co-uk** [comment]
Looks like it's duplicated as of https://github.com/doctrine/common/pull/376

File: lib/Doctrine/Common/Proxy/ProxyGenerator.php [hunk 1]
```diff
@@ -802,6 +807,23 @@ private function generateMethods(ClassMetadata $class)
...
+     *
+     * @param ReflectionMethod $method
+     *
+     * @return string
+     */
+    protected function getReturnType(\ReflectionMethod $method){
+        $returnType = $method->getReturnType();
+
+        if (! $returnType->isBuiltin()) {
+            $returnType = '\\' . $returnType;
```
Comment (stof) [inline lib/Doctrine/Common/Proxy/ProxyGenerator.php]: this will not handle ``self`` properly, will it ?

**Comment by Majkl578** [comment]
This is indeed duplicate of #376.

**Comment by Ocramius** [comment]
Dupe of #376 indeed. Closing

**Pull request closed without merging**